code/modules/admin/species_ban.dm

```diff
@@ -0,0 +1,107 @@
+// Species ban system.
+// Disallows a player from using a specific species, but lets them play the game regardless
+// Useful if someone isnt playing a species properly at all (IE: Not following hard-driven species lore)
+
+/// Global list of all species bans. Key is a ckey, value is a list of species. Its a global list to avoid mass DB lookups.
+GLOBAL_LIST_EMPTY(species_bans_assoc)
+//GLOBAL_PROTECT(species_bans_assoc) // Mainly just to save space in the already bloated VV window
+
+/datum/admins/proc/add_species_ban(ckey, species)
+	if(!ckey || !species)
+		return
+	if(!GLOB.species_bans_assoc["[ckey]"]) // Incase someone has a ckey thats just all numbers
```

Comment:
I did not think it was possible to have an all-numbers ckey?

Reply:
Better safe than sorry